pkg/test/framework/components/echo/workload.go

```diff
@@ -66,6 +66,26 @@ func (ws Workloads) Len() int {
 	return len(ws)
 }
 
+// Addresses returns the list of addresses for all workloads.
+func (ws Workloads) Addresses() []string {
+	out := make([]string, 0, len(ws))
+	for _, w := range ws {
+		out = append(out, w.Address())
+	}
+	return out
+}
+
+// FormattedAddresses is a utility function that returns a formatted version of the Addresses
```

Comment:
nit: slightly skeptical this should be used rather than something like `{{.FormattedAddresses|toYaml|indent x}}` or something?

Reply:
oh especially if its like `notValues: [{{ .ipA }}, {{ .ipB }}, {{ .ipC }}]`. That could definitely be improved